### PR TITLE
Add functionality to repeat last callout upon shaking device + RCOS Documentation

### PR DESCRIPTION
**Pull request by ConnorTheDemon**

### CODE REVIEW

**Comment by RDMurray** [comment]
Hi. Thanks for working on this! It still needs quite a bit of clean up though; there are remnants of previous attempts, commented sections of code, whitespace changes in unrelated files etc. 

I also think a setting to optionally enable this is essential. It can be very easy to trigger a shake action by mistake especially when moving around.

Please don't be discouraged by this comment - we really appreciate your work on this. Your notes are an interesting read and clarify some things which aren't well explained in the existing documentation.

File: apps/ios/GuideDogs/Code/Visual UI/View Controllers/POI Table/SearchResultsTableViewController.swift
Comment (steinbro) [inline apps/ios/GuideDogs/Code/Visual UI/View Controllers/POI Table/SearchResultsTableViewController.swift:1]: All changes in this file are extraneous.

File: apps/ios/GuideDogs/Code/Behaviors/Default/Callouts/POICallout.swift
Comment (steinbro) [inline apps/ios/GuideDogs/Code/Behaviors/Default/Callouts/POICallout.swift:1]: All changes in this file are extraneous.

File: apps/ios/GuideDogs/Code/Behaviors/Helpers/CalloutStateMachine.swift
Comment (steinbro) [inline apps/ios/GuideDogs/Code/Behaviors/Helpers/CalloutStateMachine.swift:1]: All changes in this file are extraneous.

File: apps/ios/GuideDogs/Code/Behaviors/Helpers/CalloutStateMachine.swift [hunk 1]
```diff
@@ -337,32 +338,38 @@ extension CalloutStateMachine {
...
-                
+
                 guard success else {
                     GDLogVerbose(.stateMachine, "Callout did not finish playing successfully. Terminating state machine...")
                     calloutGroup.onComplete?(false)
                     strongSelf.stateMachine.fireEvent(.failed)
                     return
                 }
                 
-                strongSelf.stateMachine.fireEvent(.delayCalloutAnnounced)
```
Comment (steinbro) [inline apps/ios/GuideDogs/Code/Behaviors/Helpers/CalloutStateMachine.swift:363]: Except maybe this deletion is meaningful?

File: apps/ios/GuideDogs/Code/Data/Authored Activities/AuthoredActivityLoader.swift [hunk 2]
```diff
@@ -505,7 +505,7 @@ class AuthoredActivityLoader {
             
             for (wptIndex, waypoint) in content.waypoints.enumerated() {
                 for (clipIndex, image) in waypoint.images.enumerated() {
-                    guard let key = manager.cacheKey(for: image.url), await manager.imageCache.containsImage(forKey: key, cacheType: .all) == .none else {
+                    guard let key = manager.cacheKey(for: image.url), await manager.imageCache.containsImage!(forKey: key, cacheType: .all) == .none else {
```
Comment (steinbro) [inline apps/ios/GuideDogs/Code/Data/Authored Activities/AuthoredActivityLoader.swift:508]: Why was the exclamation point added here? Is it necessary?

File: apps/ios/GuideDogs/Code/Data/Models/JSON Parsing/OSM/GeoJsonGeometry.swift [hunk 3]
```diff
@@ -318,7 +318,7 @@ public enum GeoJsonGeometry: Equatable, Codable {
 
 extension GeoJsonGeometry {
     private static func into_coord_pair(_ coord: CLLocationCoordinate2D) -> [CLLocationDegrees] {
-        return [coord.longitude, coord.latitude]
+        return [coord.latitude, coord.longitude]
```
Comment (steinbro) [inline apps/ios/GuideDogs/Code/Data/Models/JSON Parsing/OSM/GeoJsonGeometry.swift:321]: @RDMurray is this the latitude/longitude swap in JSON parsing that you caught? Which one was correct?

**Review by steinbro — COMMENTED** [review]
Thanks for the cleanup! I haven't tested it, but I did note a few files that don't appear to have meaningful changes so should probably be reverted before merging. I also left a couple questions for you.

**Comment by ConnorTheDemon** [comment]
For many of the changes, I don't know how they came to fruition (things like the changes in Packages.resolved, Map.storyboard, and exclamation marks added to a few of the files). Maybe they somehow got added when I pulled the branch (I remember way back when starting this project that there were a few compilation errors due to running this on a newer version of XCode, so maybe they got added there?). I wonder if these changes should be reverted or not before pulling this. And I'll work on removing the extraneous information - they generally seem to be things I missed in my first run-through of cleaning up the program.

File: apps/ios/GuideDogs/Code/Behaviors/Helpers/CalloutStateMachine.swift (hunk 1 above)
Comment (ConnorTheDemon) [reply apps/ios/GuideDogs/Code/Behaviors/Helpers/CalloutStateMachine.swift:363]: (Based on the comment below) I don't know how this ended up getting deleted. I experimented a lot with the code when I was just getting started, so maybe I deleted this in that process. I have re-added it, so you should see the line added back on the next push of this branch.

**Comment by ConnorTheDemon** [comment]
Here's the new commit. It should be virtually ready to be pulled (after resolving the issues listed above). Another thing: I question how the "allow callouts" toggle actually works. It seems like all the buttons below are functional when callouts are disabled. My code that supports shaking the device to repeat the last callout also still works. If you know how this toggle is supposed to work, please reply down here and I can make changes accordingly.

**Comment by steinbro** [comment]
Finally merged! Thanks for tackling this.

**Pull request merged**